CMakeLists.txt

```diff
@@ -162,6 +162,28 @@ IF (DEAL_II_STATIC_EXECUTABLE STREQUAL "ON")
   MESSAGE(STATUS "Creating a statically linked executable")
   SET(ASPECT_USE_SHARED_LIBS OFF CACHE BOOL "" FORCE)
 ENDIF()
+
+INCLUDE (CheckCXXSourceCompiles)
+
+SET(_backup_libs ${CMAKE_REQUIRED_LIBRARIES})
+LIST(APPEND CMAKE_REQUIRED_LIBRARIES ${CMAKE_DL_LIBS})
```

Comment:
Is `CMAKE_DL_LIBS` something cmake defines automatically for us? Is there a module that needs to be run to determine what this variable requires? What is its value for you?